tinker/configuration.py

```diff
@@ -64,6 +77,55 @@ def iterate_generator(iterates: List[ConfigEntry]) -> Iterator[ConfigEntry]:
     return itertools.chain(*(config_generator(i) for i in iterates))
 
 
+def is_smqtk(value: Any) -> bool:
+    """Return `True` if `value` is a "smqtk" directive."""
+    return type(value) is dict and "smqtk" in value
+
+
+def smqtk_generator(smqtk_def: Dict[str, ConfigEntry]) -> Iterator[ConfigEntry]:
```

Comment:
> My vision for it is that the meta-configuration expresses where it wants a SMQTK object, and can supply the class name to instantiate.

In your vision, who owns the primary declaration ability? It sounds so far that the direction is for the configuration to own the primary ability to declare parameters and values for those parameters (i.e. "expresses where it wants a SMQTK object"). This sounds like it would be somewhat in conflict with specific Protocol implementations have specific needs that need to be fulfilled that the configuration has the ability to fly in the face of. SMQTK `Configurable` takes the stance where a class implementation holds the responsibility of declaring configuration parameterization (via constructor introspection) and validation (the `from_config` classmethod, which would recursively handle nested Pluggable-/Configurable-type hydration).

One reason I'm trying to clearly understand what your vision is here is because I would love to support your use-case via helpers *in* smqtk-core. Configuration and hydration of Pluggable/Configurable types is part of what smqtk-core is all about.

Reply:
I'm not sure if this answers your question, but the same person would generally write a protocol and the configuration for some experiment that runs according to that protocol.

In the most basic fallback case, the configuration could just contain strings naming, say, different algorithms to use. The protocol could, e.g., translate these names to SMQTK objects at runtime, then use those. The `smqtk` directive is there simply to short-circuit this hand-translation of names into SMQTK objects. In accordance with that use case, we're thinking about how to express some general constraints directly in the configuration file that could help to control what gets passed into the runtime protocol.

So, I'm thinking about a [job to be done](https://hbr.org/2016/09/know-your-customers-jobs-to-be-done) here; if SMQTK native utilities can do that job better than this vision for the `smqtk` directive in Tinker Engine configurations, I'd like to learn about them and use them. I'm purposefully keeping my mental model of this system simple, so I do think we need a live conversation to explore some of the intricacies that may be possible here with SMQTK.